From f624b046bafb4ce22f43dbd13e34adfa83380774 Mon Sep 17 00:00:00 2001
From: Quentin Casasnovas <quentinx.casasnovas@intel.com>
Date: Tue, 18 Dec 2012 16:59:53 +0100
Subject: [PATCH 71/78] wl12xx: Reduce command timeout from 2secs to 250ms

BZ: 75240

During the suspend path, the power management subsystem from within the kernel
will bug on if the suspend path takes more than 3 seconds. The wl12xx may have
a firmware recovery parallel to the suspend path (700ms), plus the 2 seconds it
was taking to timeout a command, we were getting very close of this 3 seconds
limit. In certain rare corner cases, the limit is reach and the platform just
reboots.

This patch changes the COMMAND_TIMEOUT to 250ms. On a sample of 3000 commands,
we've found an average time to complete the command of 3msec (max was 25ms on 8
commands out of 3000). 250ms is 10 times the max observed, so this should give
us much better reactivity and prevents the DPM timer from rebooting the device.

Change-Id: I67f9568caa9658d9ecd647adbe35d784b03dedee
Signed-off-by: Quentin Casasnovas <quentinx.casasnovas@intel.com>
Reviewed-on: http://android.intel.com:8080/82500
Reviewed-by: Zayet, Aymen <aymen.zayet@intel.com>
Reviewed-by: Garcia, JeremieX <jeremiex.garcia@intel.com>
Reviewed-by: De Cesco, Jonathan <jonathan.de.cesco@intel.com>
Tested-by: Zurmely, PierreX <pierrex.zurmely@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 wl12xx-compat/drivers/net/wireless/wl12xx/cmd.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/wl12xx-compat/drivers/net/wireless/wl12xx/cmd.h b/wl12xx-compat/drivers/net/wireless/wl12xx/cmd.h
index 3a67fe6..154a70e 100644
--- a/wl12xx-compat/drivers/net/wireless/wl12xx/cmd.h
+++ b/wl12xx-compat/drivers/net/wireless/wl12xx/cmd.h
@@ -194,7 +194,7 @@ enum cmd_templ {
 };
 
 /* unit ms */
-#define WL1271_COMMAND_TIMEOUT     2000
+#define WL1271_COMMAND_TIMEOUT     250
 #define WL1271_CMD_TEMPL_DFLT_SIZE 252
 #define WL1271_CMD_TEMPL_MAX_SIZE  512
 #define WL1271_EVENT_TIMEOUT       1500
-- 
1.7.9

